### PR TITLE
Add link to CO and Taxon for preparations in interaction dialog

### LINKED ISSUE #2718: Add links to CO and preparation records from the interaction selection dialog
image

Matches the feature in Specify 6

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #2718

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx [hunk 1]
```diff
@@ -173,14 +173,16 @@ export function InteractionDialog({
...
         catalogNumber: prepData[0],
-        taxon: prepData[1],
-        preparationId: prepData[2],
-        prepType: prepData[3],
-        countAmount: prepData[4],
-        loaned: f.parseInt(prepData[5] ?? undefined) ?? 0,
-        gifted: f.parseInt(prepData[6] ?? undefined) ?? 0,
-        exchanged: f.parseInt(prepData[7] ?? undefined) ?? 0,
-        available: Number.parseInt(prepData[8]),
+        coId: prepData[1],
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx:176]: Replace with `collectionObjectId`. Please don't use contraction in this codebase in the future it's not always clear what was meant by it.

Case in point, do you have an idea of what our previous engineer meant by `LDLM` in `LDLM_TO_MYSQL`? No other references to it in the codebase or online

File: specifyweb/frontend/js_src/lib/components/Interactions/PrepDialogRow.tsx [hunk 2]
```diff
@@ -64,14 +65,41 @@ export function PrepDialogRow({
...
-            getField(schema.models.CollectionObject, 'catalogNumber'),
-            undefined,
-            preparation.catalogNumber
-          )}
-        </td>
-        <td>{preparation.taxon}</td>
+        {typeof preparation.coId === 'number' ? (
+          <td className="justify-end tabular-nums">
+            <Link.NewTab
+              href={`/specify/view/collectionobject/${preparation.coId}`}
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Interactions/PrepDialogRow.tsx:71]: Use `getResourceViewUrl('CollectionObject', preparation.collectionObjectId)` instead.
Reason: it's easy to make a typo in a string or forget a `/`, but when calling getResourceViewUrl everything is safe:
1. TypeScrpipt checks that getResourceViewUrl is a correct name of the function
2. TypeScript checks that 'CollectionObject' is a valid table name
3. TypeScript checks that preparation.collectionObjectId is a number
4. In the future, if we want to change the URL, we just have to change the function and all the places are updated automatically!

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]

**Review by maxxxxxdlp — APPROVED** [review]

**Comment by realVinayak** [comment]
https://user-images.githubusercontent.com/61122018/228662733-a8535808-0293-4ccb-a7bf-501b64547143.mp4

Clicking on record set doesn't open interactions dialog at all.

**Comment by CarolineDenis** [comment]
> Clicking on record set doesn't open interactions dialog at all.

This behavior was the behavior previous to this PR. You cannot 'click'/open a record set if it doesn't contain any preparation to loan.

**Comment by realVinayak** [comment]
Ah, thanks for clearing that up for me!

**Review by grantfitzsimmons — APPROVED** [review]
I wouldn't merge yet as we go into the next couple of releases, but post-`statspage` I am excited to see this feature!

Works great!

**Pull request merged**